casbin/config/config.cpp

```diff
@@ -0,0 +1,186 @@
+#include "config.h"
+#include "../util/util.h"
+#include <iostream>
+#include <sstream>
+#include <algorithm>
+#include <fstream>
+
+const string DEFAULT_SECTION = "default";
+const string DEFAULT_COMMENT = "#";
+const string DEFAULT_COMMENT_SEM = ";";
+const string DEFAULT_MULTI_LINE_SEPARATOR = "\\";
+
+
+
+Config::Config() {
+
+}
+
+Config* Config::NewConfigFromFile(Error& err,const string& path) {
+	Config* cfg = new Config();
+	err = cfg->parseFile(path);
+	return cfg;
+}
+
+Config* Config::NewConfigFromText(Error& err, const string& text) {
+	Config* cfg = new Config();
+	err = cfg->parseText(text);
+	return cfg;
+}
+
+Error Config::parseFile(const string& fname) {
+	m.lock();

```

Comment:
No need to add lock here, data races are unlikely to happen here.

Reply:
ok